run.py

```diff
@@ -197,6 +197,106 @@ def get_files(args, dirname, min_size=1, min_date=''):
    return set(ls_s3_dir("%s/%s/%s/" % (S3_BUCKET, args.bioproject, dirname),
                         min_size=min_size, min_date=min_date))
 
+def ribocounts(args):
+    """Count the number of reads classified as non-rRNA by RiboDetector"""
+    calc_avg_read_length = False
+
+    available_inputs = get_files(args, "cleaned",
+                                 # tiny files are empty; ignore them
+                                 min_size=100)
+    existing_outputs = get_files(args, "ribocounts")
+        
+    for sample in get_samples(args):
+        # Check for name of output file
+        sample_output_file = sample + ".ribocounts.txt"
+        if sample_output_file in existing_outputs: continue
+
+        sample_nonrrna_count = 0
+        for potential_input in available_inputs:
+            if not potential_input.startswith(sample): continue
+            if ".settings" in potential_input: continue
+            if "discarded" in potential_input: continue
+
+            # Number of output and input files must match 
+            tmp_fq_outputs = potential_input.replace(".gz", ".nonrrna.fq")
+            tmp_fq_outputs_list = [tmp_fq_outputs]
+            inputs = [potential_input]
+
+            if ".pair1." in potential_input:
+                tmp_fq_outputs_list.append(tmp_fq_outputs.replace(".pair1.", ".pair2."))
+                inputs.append(potential_input.replace(".pair1.", ".pair2."))
+            elif ".pair2." in potential_input:
+                # Ribodetector handles pair1 and pair2 together.
+                continue
+
+            with tempdir("ribocounts", sample + " inputs") as workdir:
+                for input_fname in inputs:
+                    subprocess.check_call([
+                        "aws", "s3", "cp", "%s/%s/cleaned/%s" % (
+                            S3_BUCKET, args.bioproject, input_fname), input_fname])
+
+                    # Ribodetector gets angry if the .fq extension isn't in the filename 
+                    os.rename(input_fname, input_fname.replace(".gz", ".fq.gz"))
+
+                # Add .fq extensions to input files
+                inputs = [i.replace(".gz", ".fq.gz") for i in inputs]
+
+                # Optional: compute average read lengths. Note that if we implement this option we can count total reads here. For paired-end reads, average length is computed only from pair1 reads.
+                if calc_avg_read_length:
+                    print("Calculating average read length...")
+                    def calculate_average_read_length(file_path):
```

Comment:
Don't parse FASTQ manually in python; the format is a bit more subtle than this (ex: it allows wrapping long lines).  How about:

```
total_len = 0
total_reads = 0
with gzip.open(file_path, 'rt') as inf:
  for (title, sequence, quality) in FastqGeneralIterator(inf):
     total_len += len(sequence)
     total_reads += 1
return total_reads, round(total_len / total_reads)
```